otelcourier/publish_test.go

```diff
@@ -95,6 +84,38 @@ func TestPublishTraceSpan(t *testing.T) {
 		assert.Equal(t, publishErrMessage, span.Status().Description)
 	})
 
+	t.Run("Metrics", func(t *testing.T) {
+		vsn := courier.Version()
+		buf := bytes.NewBufferString(fmt.Sprintf(`# HELP courier_publish_attempts_total Number of publish attempts
+# TYPE courier_publish_attempts_total counter
+courier_publish_attempts_total{mqtt_qos="1",mqtt_retained="false",mqtt_topic="test-topic",otel_scope_name="github.com/gojek/courier-go/otelcourier",otel_scope_version="semver:%s",service_name="test-service"} 1
```

Comment:
mqtt_topic label can have non-deterministic values which can break the prometheus limit.

Reply:
Same goes for subscribe and unsubscribe events.

Reply:
Addressed with a new `Option`.

```go
// TopicAttributeTransformer helps transform topic before making an attribute for it.
// It is used in metric recording only. Traces use the original topic.
type TopicAttributeTransformer func(context.Context, string) string
```